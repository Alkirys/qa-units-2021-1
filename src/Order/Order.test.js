import React from 'react';
jest.mock('../utils/getDate');
import {getDate} from '../utils/getDate';
import {fakeOrders} from '../data/fakeOrders';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Order from './Order';

configure({ adapter: new Adapter() });

describe('Order.js', () => {
  beforeEach(() => {
    jest.resetModules();
    getDate.mockReturnValue('31 февраля, пн, 2021 год');
  });

  it('all right', () => {
    const wrapper = shallow(<Order order={fakeOrders[0]}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('no order', () => {
    const wrapper = shallow(<Order />);
    expect(wrapper.getElement()).toBeNull();
  });

  it('order is empty', () => {
    const wrapper = shallow(<Order order={{}}/>);
    expect(wrapper.getElement()).toBeNull();
  });

  it('items is empty', () => {
    const order = Object.assign({}, fakeOrders[0]);
    order.items = undefined;
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper).toMatchSnapshot();
  });
});

