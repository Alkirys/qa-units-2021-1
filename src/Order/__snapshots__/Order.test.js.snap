// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js all right 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      31 февраля, пн, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order.js no order 1`] = `""`;

exports[`Order.js no shop, date in order 1`] = `""`;
